sklearn/cluster/tests/test_bisect_k_means.py

```diff
@@ -133,3 +133,17 @@ def test_float32_float64_equivalence(csr_container):
 
     assert_allclose(km32.cluster_centers_, km64.cluster_centers_)
     assert_array_equal(km32.labels_, km64.labels_)
+
+
+def test_non_regression_segfault_on_empty_bisections():
+    # Non-regression test for:
+    # https://github.com/scikit-learn/scikit-learn/issues/27081
+    rng = np.random.RandomState(0)
+    X_train = rng.rand(3000, 10)
+    bkm = BisectingKMeans(n_clusters=10).fit(X_train)
```

Comment:
To also test for `algorithm="elkan"`, taking https://github.com/scikit-learn/scikit-learn/pull/27167/files#r1305929679 into account:
```suggestion
@pytest.mark.parametrize("algorithm", ("lloyd", "elkan")):
def test_no_crash_on_empty_bisections(algorithm):
    # Non-regression test for:
    # https://github.com/scikit-learn/scikit-learn/issues/27081
    rng = np.random.RandomState(0)
    X_train = rng.rand(3000, 10)
    bkm = BisectingKMeans(n_clusters=10, algorithm=algorithm).fit(X_train)
```